xbrowse/annotation/vep_annotations.py

```diff
@@ -257,47 +257,64 @@ def get_csq_fields_from_vcf_desc(csq_desc):
 
 
 
-def get_worst_vep_annotation_index(vep_annotation, gene_id=None):
+def get_worst_vep_annotation_index(transcript_annotations, gene_id=None):
     """
-    Returns index of which VEP annotation is worst (zero-indexed)
-    Exception if no vep annotation for some reason
-
-    if you want the index of worst annotation for a given gene, pass gene_id
-    gene_id is None implies the worst global annotation
+    Returns (zero-based) index of the VEP annotation with the worst consequence.
 
+    Args:
+        transcript_annotations: a list where each element represents VEP annotations
+            for a different transcript (parsed from a VCF record's CSQ field).
+        gene_id: if specified, only annotations with this gene_id will be
+            considered. This is useful when the transcript_annotations list
+            contains transcripts from more than one gene.
     """
 
-    num_annotations = len(vep_annotation)
-    if num_annotations == 0:
-        print 'Warning: no VEP annnotation'
-        return None
+    if not transcript_annotations:
+        raise ValueError("transcript_annotations is empty")
 
-    worst_value = 1000
-    worst_index = -1
-    for i in range(num_annotations):
```

Comment:
xrange is a nice function to use here as well (vs range) with caveats. Good pros/cons listed here http://stackoverflow.com/questions/135041/should-you-always-favor-xrange-over-range